# Command Line Tools for Genomic Data Science Grade 4
## Grade 4

### Question 1
How many alignments were produced for the ‘Day8’ RNA-seq data set?

    DATADIR=/home/guest/Downloads/gencommand_proj4
    WORKDIR=/home/guest/Downloads/gencommand_proj4
    ANNOT=/home/guest/Downloads/gencommand_proj4/athal_genes.gtf
    ANNOTIDX=/home/guest/Downloads/gencommand_proj4/athal_genes_bt2index/athal_genes_bt2index
    BWT2IDX=/home/guest/Downloads/gencommand_proj4/athal/athal

    mkdir -p $WORKDIR/Test1/Day8

    tophat2 -o $WORKDIR/Test1/Day8 -p 10 \
        $BWT2IDX \
        $DATADIR/Day8.fastq
    samtools view Test1/Day8/accepted_hits.bam|wc -l
    
ANS

    63845
### Question 2
How many alignments were produced for the ‘Day16’ RNA-seq data set?

    DATADIR=/home/guest/Downloads/gencommand_proj4
    WORKDIR=/home/guest/Downloads/gencommand_proj4
    ANNOT=/home/guest/Downloads/gencommand_proj4/athal_genes.gtf
    ANNOTIDX=/home/guest/Downloads/gencommand_proj4/athal_genes_bt2index/athal_genes_bt2index
    BWT2IDX=/home/guest/Downloads/gencommand_proj4/athal/athal

    mkdir -p $WORKDIR/Test1/Day8

    tophat2 -o $WORKDIR/Test1/Day8 -p 10 \
        $BWT2IDX \
        $DATADIR/Day8.fastq
    samtools view Test1/Day8/accepted_hits.bam|wc -l
ANS

    58398
### Question 3
How many reads were mapped in ‘Day8’ RNA-seq data set?
    
    samtools view Test1/Day8/accepted_hits.bam|cut -f 1|sort -u|wc -l
Ans
    
    63489
    
### Question 4
Question 4
How many reads were mapped in ‘Day16’ RNA-seq data set?

    samtools view Test1/Day8/accepted_hits.bam|cut -f 1|sort -u|wc -l
Ans
    
    57951

### Question 5
How many reads were uniquely aligned in ‘Day8’ RNA-seq data set?

    samtools view Test1/Day8/accepted_hits.bam|cut -f 1|uniq -c |grep "1 SRR" |wc -l

Ans:

    63133
### Question 6
How many reads were uniquely aligned in ‘Day16’ RNA-seq data set

    samtools view Test1/Day16/accepted_hits.bam|cut -f 1|uniq -c |grep "1 SRR" |wc -l
Ans:

    57504
### Question 7
How many spliced alignments were reported for ‘Day8’ RNA-seq data set
     
    samtools view Test1/Day8/accepted_hits.bam |cut -f6 |grep N |wc -l
Ans

    8596
### Question 8
How many spliced alignments were reported for ‘Day16’ RNA-seq data set?
    
    samtools view Test1/Day16/accepted_hits.bam |cut -f6 |grep N |wc -l
Ans

    10695
    
### Question 9
How many reads were left unmapped from ‘Day8’ RNA-seq data set?

    samtools view Test1/Day16/unmapped.bam |cut -f1|sort -u|wc -l
Ans

    84

### Question 10
How many reads were left unmapped from ‘Day16’ RNA-seq data set?

    samtools view Test1/Day8/unmapped.bam |cut -f1|sort -u|wc -l
Ans
    
    34    

### Question 11
How many genes were generated by cufflinks for Day8?

     THDIR=/home/guest/Downloads/gencommand_proj4/Test1/Day8/
    WORKDIR=/home/guest/Downloads/gencommand_proj4/Test1/Day8/

    cd $WORKDIR
    cufflinks -L BJ-A0Z2  $THDIR/accepted_hits.bam
    cut -f4 Test1/Day8/genes.fpkm_tracking |wc -l
Ans

    186


### Question 12
How many genes were generated by cufflinks for Day16?

    THDIR=/home/guest/Downloads/gencommand_proj4/Test1/Day16/
    WORKDIR=/home/guest/Downloads/gencommand_proj4/Test1/Day16/

    cd $WORKDIR
    cufflinks -L BJ-A0Z2  $THDIR/accepted_hits.bam
    cut -f4 Test1/Day16/genes.fpkm_tracking |wc -l
Ans

    80


### Question 13
How many transcripts were reported for Day8?

    cut -f1 Test1/Day8/isoforms.fpkm_tracking|wc -l
Ans

    192

### Question 14
How many transcripts were reported for Day16?

    cut -f1 Test1/Day16/isoforms.fpkm_tracking|wc -l
Ans

    92


### Question 15
How many single transcript genes were produced for Day8?

    cut -f1 Test1/Day8/isoforms.fpkm_tracking|grep ".1$" |wc -l
    cut -f1 Test1/Day8/isoforms.fpkm_tracking|grep ".2$" |wc -l
Ans

    180

### Question 16
How many single transcript genes were produced for Day16?

    cut -f1 Test1/Day16/isoforms.fpkm_tracking|grep ".1$" |wc -l
    cut -f1 Test1/Day16/isoforms.fpkm_tracking|grep ".2$" |wc -l
Ans

    69

### Question 17
How many single-exon transcripts were in the Day8 set?

    cut -f9 Test1/Day8/transcripts.gtf|grep exon_number\ \"1\"|wc -l
    cut -f9 Test1/Day8/transcripts.gtf|grep exon_number\ \"2\"|wc -l
Ans
    
    119
### Question 18
How many single-exon transcripts were in the Day16 set?

    cut -f9 Test1/Day16/transcripts.gtf|grep exon_number\ \"1\"|wc -l
    cut -f9 Test1/Day16/transcripts.gtf|grep exon_number\ \"2\"|wc -l
Ans
    
    24

### Question 19
How many multi-exon transcripts were in the Day8 set?

    cut -f9 Test1/Day8/transcripts.gtf|grep exon_number\ \"2\"|wc -l
Ans

    73

### Question 20
How many multi-exon transcripts were in the Day16 set?

    cut -f9 Test1/Day16/transcripts.gtf|grep exon_number\ \"2\"|wc -l
Ans

    68
### Question 21
How many cufflinks transcripts fully reconstruct annotation transcripts in Day8?

    cuffcompare -r athal_genes.gtf -R Test1/Day8/transcripts.gtf -o Test1/Day8
    cut -f3 Test1/Day8/Day8.transcripts.gtf.tmap |sort|uniq -c
Ans
    
    16

### Question 22
How many cufflinks transcripts fully reconstruct annotation transcripts in Day16?

    cuffcompare -r athal_genes.gtf -R Test1/Day16/transcripts.gtf -o Test1/Day16
    cut -f3 Test1/Day16/Day16.transcripts.gtf.tmap |sort|uniq -c
Ans

    36

### Question 23
How many splice variants does the gene AT4G20240 have in the Day8 sample?

    grep AT4G20240 Test1/Day8/Day8.transcripts.gtf.tmap|wc -l
Ans
    
    2
   
### Question 24
How many splice variants does the gene AT4G20240 have in the Day16 sample?

    grep AT4G20240 Test1/Day16/Day16.transcripts.gtf.tmap|wc -l
Ans
    
    0

### Question 25
How many cufflinks transcripts are partial reconstructions of reference transcripts (‘contained’)? (Day8)

    cut -f3 Test1/Day8/Day8.transcripts.gtf.tmap |sort|uniq -c    
Ans

    133

### Question 26
How many cufflinks transcripts are partial reconstructions of reference transcripts (‘contained’)? (Day16)

    cut -f3 Test1/Day16/Day16.transcripts.gtf.tmap |sort|uniq -c
Ans

    21

### Question 27
How many cufflinks transcripts are novel splice variants of reference genes? (Day8)

    cut -f3 Test1/Day8/Day8.transcripts.gtf.tmap |sort|uniq -c
Ans

    14

### Question 28
How many cufflinks transcripts are novel splice variants of reference genes? (Day16)

    cut -f3 Test1/Day16/Day16.transcripts.gtf.tmap |sort|uniq -c
Ans

    22
    
### Question 29
How many cufflinks transcripts were formed in the introns of reference genes? (Day8)

    cut -f3 Test1/Day8/Day8.transcripts.gtf.tmap |sort|uniq -c
    
Ans

    4

### Question 30
How many cufflinks transcripts were formed in the introns of reference genes? (Day16)

    cut -f3 Test1/Day16/Day16.transcripts.gtf.tmap |sort|uniq -c
    
Ans

    1

### Question 31
How many genes (loci) were reported in the merged.gtf file?
    
    WORKDIR=/home/guest/Downloads/gencommand_proj4
    THDIR=/home/guest/Downloads/gencommand_proj4/Test1
    ANNOT=/home/guest/Downloads/gencommand_proj4/athal_genes.gtf

    cuffmerge -g $ANNOT -o $WORKDIR/Test1/Cuffmerge $WORKDIR/commands/GTFs.txt

    cuffdiff -o $WORKDIR/Test1/CuffDiff $WORKDIR/Test1/Cuffmerge/merged.gtf \
          $THDIR/Day8/accepted_hits.bam $THDIR/Day16/accepted_hits.bam

    cut -f9 Test1/Cuffmerge/merged.gtf|cut -f2 -d " "|sort -u|wc -l

Ans

    129
### Question 32
How many transcripts?

    cut -f9 Test1/Cuffmerge/merged.gtf|cut -f4 -d " "|sort -u|wc -l
Ans
   
    200

### Question 33
How many genes total were included in the gene expression report from cuffdiff?

    cat Test1/CuffDiff/gene_exp.diff|wc -l
  
Ans

    129
### Question 34
How many genes were detected as differentially expressed?

    cat Test1/CuffDiff/gene_exp.diff|cut -f14|grep yes|wc -l
Ans
    
    4
### Question 35
How many transcripts were differentially expressed between the two samples?

    cat Test1/CuffDiff/isoform_exp.diff|cut -f 14|grep yes|wc -l
Ans

    5
